drivers/spi/spi_renesas_ra.c

```diff
@@ -0,0 +1,826 @@
+			.transfer_settings_word_b.size = TRANSFER_SIZE_1_BYTE,                     \
+			.transfer_settings_word_b.mode = TRANSFER_MODE_NORMAL,                     \
+			.p_dest = (void *)NULL,                                                    \
+			.p_src = (void const *)NULL,                                               \
+			.num_blocks = 0,                                                           \
+			.length = 0,                                                               \
+	},                                                                                         \
+	.tx_transfer_cfg_extend = {.activation_source = DT_INST_IRQ_BY_NAME(index, txi, irq)},     \
+	.tx_transfer_cfg =                                                                         \
+		{                                                                                  \
+			.p_info = &ra_spi_data_##index.tx_transfer_info,                           \
+			.p_extend = &ra_spi_data_##index.tx_transfer_cfg_extend,                   \
+	},                                                                                         \
+	.tx_transfer = {                                                                           \
+		.p_ctrl = &ra_spi_data_##index.tx_transfer_ctrl,                                   \
+		.p_cfg = &ra_spi_data_##index.tx_transfer_cfg,                                     \
+		.p_api = &g_transfer_on_dtc,                                                       \
+	},
+#endif
+
+#define RA_SPI_INIT(index)                                                                         \
+                                                                                                   \
+	PINCTRL_DT_INST_DEFINE(index);                                                             \
+                                                                                                   \
+	static const struct ra_spi_config ra_spi_config_##index = {                                \
+		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),                                     \
+	};                                                                                         \
+                                                                                                   \
+	static struct ra_spi_data ra_spi_data_##index = {                                          \
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(index), ctx)                           \
+			SPI_CONTEXT_INIT_LOCK(ra_spi_data_##index, ctx),                           \
+		SPI_CONTEXT_INIT_SYNC(ra_spi_data_##index, ctx),                                   \
+		.fsp_config =                                                                      \
+			{                                                                          \
+				.channel = DT_INST_PROP(index, channel),                           \
+				.rxi_ipl = DT_INST_IRQ_BY_NAME(index, rxi, priority),              \
```

Comment:
IRQ_CONNECT() may alter the priority passed. Please check z_arm_irq_priority_set() function.
Subsequent calls to  `R_SPI_Open()` will change this priority.
It will be good to update all `*_ipl` after  IRQ_CONNECT().